### ISSUE TITLE
breaking change for coeffs of Polynominal(0) ?

### ISSUE THREAD

**Issue by johnnychen94** [JuliaMath/Polynomials.jl#503]
I'm not sure if this is an intended change, will update the thread if I get more ideas on how the behavior get affected.

v3.2.8:

```julia
julia> using Polynomials

julia> coeffs(Polynomial(0))
1-element Vector{Int64}:
 0
```

v3.2.9-3.2.12:

```julia
julia> using Polynomials

julia> coeffs(Polynomial(0))
Int64[]
```

**Comment by jverzani** [comment]
Yes, there was a change here. The basic constructors always trimmed any trailing zeros, and the 
zero polynomials were special cased.  That was dropped so that `zero(T)` was not needed to make a zero polynomial. Does it effect anything that you were doing?

**Comment by johnnychen94** [comment]
> Does it effect anything that you were doing?

Yep, when implementing conv or some loops, it would occasionally depend on `length(coeffs(p))`. I've already gotten a [compat patch](https://github.com/Suzhou-Tongyuan/GaloisFieldNumbers.jl/commit/4daa9b05136f5dd85e69af6db50b9f43e7801f93) for this so not much to worry. I just opened this issue to see if there's a need to "fix something upstream" 😆
